src/csync/csync_update.cpp

```diff
@@ -96,6 +76,14 @@ static bool _last_db_return_error(CSYNC* ctx) {
     return ctx->statedb.lastReturnValue != SQLITE_OK && ctx->statedb.lastReturnValue != SQLITE_DONE && ctx->statedb.lastReturnValue != SQLITE_ROW;
 }
 
+static QByteArray _rel_to_abs(CSYNC* ctx, const QByteArray &relativePath) {
+    QByteArray abs = QByteArray::fromRawData(ctx->local.uri, strlen(ctx->local.uri));
```

Comment:
this still allocates.  and then re-allocates.

It should simply be:

```
return QByteArray() % ctx->local.uri % '/' % relativePath;
```

Let QStringBuilder do the job and avoid allocation.

Reply:
This is using `fromRawData` and I don't see why it would detach, why would it allocate?

Reply:
fromRawData still allowates memory for the QByteArrayData.  That makes the use of fromRawData quite useless, actually.

Reply:
Done